File: engines/pink/objects/cgame.cpp
#include "cgame.h"
#include "carchive.h"
#include "cruntimeclass.h"

#include "common/str.h"
#include "common/array.h"

#include "cmoduleproxy.h"

namespace Pink {
	using Common::Array;

	CGame::CGame() : path(NULL), ini(NULL) {
	}


	CGame::~CGame() {
		delete path;
		delete ini;
		delete modules;
	}

	void CGame::deserialize(CArchive &archive) {
		path = archive.readCString();
		ini = archive.readCString();
		modules = (Array<CModuleProxy *> *)archive.readCObArray();
	}

};

File: engines/pink/OrbFile/CGame.py
from CArchive import *

from CObject import *

class CGame(CObject):
    def __init__(self):
        self.path = ''
        self.ini = ''
        self.modules = []

    def readFromOrb(self, archive):
        archive.mapObject(self)
        self.path = archive.readCString()
        self.ini = archive.readCString()
        self.modules = archive.readCObArray()

    def __repr__(self):
        return "<CGame - %d objects in '%s' in '%s'>" % (len(self.modules), self.ini, self.path)

CArchive.register(CGame)

__all__ = ["CGame"]
